### PR TITLE
ARROW-12563: [C++][Gandiva] Add space,add_months and datediff functions for string

### PR DESCRIPTION
**Pull request by DileepSrigiri**
Space(int n): Return a string of n spaces
add_months(string start_date, int num_months): adds num_months to start_date and returns that date
datediff(string enddate, string startdate): Returns the number of days from startdate to enddate

### CODE REVIEW

File: cpp/src/gandiva/function_registry_timestamp_arithmetic.cc [hunk 1]
```diff
@@ -58,13 +64,15 @@ std::vector<NativeFunction> GetDateTimeArithmeticFunctionRegistry() {
       TIMESTAMP_DIFF_FN(timestampdiffMonth, {}),
       TIMESTAMP_DIFF_FN(timestampdiffQuarter, {}),
       TIMESTAMP_DIFF_FN(timestampdiffYear, {}),
+      TIMESTAMP_DIFF_FN(datediff, {}),
```
Comment (projjal) [inline cpp/src/gandiva/function_registry_timestamp_arithmetic.cc:67]: please add datediff and add_months as aliases

File: cpp/src/gandiva/precompiled/epoch_time_point.h [hunk 2]
```diff
@@ -19,6 +19,14 @@
 
 // TODO(wesm): IR compilation does not have any include directories set
 #include "../../arrow/vendored/datetime/date.h"
+#ifndef UTIL_FUNCTIONS
```
[the remark this answers is not in the document]
Comment (DileepSrigiri) [reply cpp/src/gandiva/precompiled/epoch_time_point.h]: removed it

File: cpp/src/gandiva/precompiled/timestamp_arithmetic.cc [hunk 3]
```diff
@@ -47,6 +47,28 @@ bool is_last_day_of_month(const EpochTimePoint& tp) {
   return !is_leap_year(tp.TmYear());
 }
 
+bool did_days_overflow(int year, int month, int days) {
+      int days_in_month[] = {31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};
```
[the remark this answers is not in the document]
Comment (DileepSrigiri) [reply cpp/src/gandiva/precompiled/timestamp_arithmetic.cc]: Done

**Comment by projjal** [comment]
The CI builds are failing.

**Review by projjal — APPROVED** [review]

**Comment by DileepSrigiri** [comment]
Functions were already added

**Pull request closed without merging**